processor/src/main/java/org/mapstruct/ap/internal/model/common/BuilderType.java

```diff
@@ -107,11 +107,6 @@ public static BuilderType create(BuilderInfo builderInfo, Type typeToBuild, Type
         if ( builderInfo == null ) {
             return null;
         }
-        ExecutableElement buildMethod = builderInfo.getBuildMethod();
```

Comment:
is this check obsolete?

Reply:
Probably this check belongs to the `BuilderFinisherMethodResolver` now. As we can check the method return type there. Although this can't happen with the default builder provider as we have this check there.

Reply:
ok